pandas/core/generic.py

```diff
@@ -4581,38 +4580,61 @@ def pipe(self, func, *args, **kwargs):
 
     Parameters
     ----------
-    func : callable, string, dictionary, or list of string/callables
-        To apply to column
+    func : function, string, list of string/functions or dictionary
+        Function to use for transforming the data. If a function, must either
+        work when passed a %(klass)s or when passed to %(klass)s.apply.
 
-        Accepted Combinations are:
+        Accepted combinations are:
 
         - string function name
         - function
-        - list of functions
-        - dict of column names -> functions (or list of functions)
+        - list of functions and/or function names
+        - dict of axis labels -> functions, function names or list of such
+    %(axis)s
+    *args
+        Positional arguments to pass to `func`.
+    **kwargs
+        Keyword arguments to pass to `func`.
 
     Returns
     -------
     transformed : %(klass)s
 
+    Raises
+    ------
+    ValueError: if the returned %(klass)s has a different length than self.
+
     Examples
     --------
-    >>> df = pd.DataFrame(np.random.randn(10, 3), columns=['A', 'B', 'C'],
+    >>> df = pd.DataFrame({'A': range(10), 'B': range(10, 0, -1)},
     ...                   index=pd.date_range('1/1/2000', periods=10))
-    df.iloc[3:7] = np.nan
+    >>> df.iloc[3:7] = np.nan
 
     >>> df.transform(lambda x: (x - x.mean()) / x.std())
-                       A         B         C
-    2000-01-01  0.579457  1.236184  0.123424
-    2000-01-02  0.370357 -0.605875 -1.231325
-    2000-01-03  1.455756 -0.277446  0.288967
-    2000-01-04       NaN       NaN       NaN
-    2000-01-05       NaN       NaN       NaN
-    2000-01-06       NaN       NaN       NaN
-    2000-01-07       NaN       NaN       NaN
-    2000-01-08 -0.498658  1.274522  1.642524
-    2000-01-09 -0.540524 -1.012676 -0.828968
-    2000-01-10 -1.366388 -0.614710  0.005378
+                       A         B
+    2000-01-01 -1.143001  1.143001
+    2000-01-02 -0.889001  0.889001
+    2000-01-03 -0.635001  0.635001
+    2000-01-04       NaN       NaN
+    2000-01-05       NaN       NaN
+    2000-01-06       NaN       NaN
+    2000-01-07       NaN       NaN
+    2000-01-08  0.635001 -0.635001
+    2000-01-09  0.889001 -0.889001
+    2000-01-10  1.143001 -1.143001
+
+    It is only required for the axis specified in the ``axis`` parameter
+    to have the same length for output and for self. The other axis may have a
+    different length:
+
+    >>> s = pd.Series(range(5))
+    >>> s.transform([np.sqrt, np.exp])
+           sqrt        exp
+    0  0.000000   1.000000
+    1  1.000000   2.718282
+    2  1.414214   7.389056
+    3  1.732051  20.085537
+    4  2.000000  54.598150
 
     See also
```

Comment:
`See Also` should be placed before the examples, and should have a capital `A` in `Also`.